sdks/python/apache_beam/runners/common.py

```diff
@@ -69,7 +69,8 @@ def __init__(self,
                logging_context=None,
                # Preferred alternative to context
                # TODO(robertwb): Remove once all runners are updated.
-               state=None):
+               state=None,
+               scoped_metrics_container=None):
```

Comment:
What if this is None (e.g. for other runners like the dataflow runner?) It seems in that case the (unconditional) self.scoped_metrics_container.enter()s below would fail. Seems we should be instantiating a no-op one in that case.